demos/composition/cyclow/es6/src/confirm.js

```diff
@@ -0,0 +1,34 @@
+import {Block} from 'cyclow'
+
+const viewConfirm = () => [
+  {tag: 'button', content: 'Confirm', on: {click: 'confirm'}},
+  {tag: 'button', content: 'Cancel', on: {click: 'cancel'}}
+]
+
+const viewDefault = mode => ({
+  tag: 'button',
+  content: 'Submit',
+  attrs: {disabled: mode === 'disabled'},
+  on: {click: 'maybe'}
+})
+
+const Confirm = () => Block({
+  inputs: ['disabled'],
+  outputs: ['confirmation'],
+  on: {
+    init: () => state => ({mode: 'disabled'}),
+    disabled: (disabled, {mode}) => {
+      if (disabled) {
+        return state => ({mode: 'disabled'})
+      } else if (mode === 'disabled') {
+        return state => ({mode: 'waiting'})
+      }
+    },
+    maybe: () => state => ({mode: 'confirm'}),
```

Comment:
Since it is not uncommon that the new state doesn't depend on the previous one, maybe Cyclow (or rather Graflow, I suspect) could allow a shorthand where instead of a `newStateMakerFunc` I simply pass `newState`, which the framework would wrap in a function for me?

In other words, allowing me to type this:

```
maybe: () => ({mode: 'confirm'})
```

...instead of this:

```
maybe: () => state => ({mode: 'confirm'}),
```